Update keyconcepts.md

altayhunoglu opened the pull request:
Group Dynamics.

ErdemT09 reviewed (commented):
https://www.mindtools.com/pages/article/improving-group-dynamics.htm
https://www.yourarticlelibrary.com/management/group-dynamics-its-characteristics-stages-types-and-other-details-management/5363
Here are two more resources that I find to be well-written.

altayhunoglu commented:
Thanks, Erdem. I added the resources you share and updated the code.

I think Group Dynamics deserves its own module, but for now, I added it as a section.
